mantidproject/mantid
# Support for legacy file format in DNS Reduction GUI

### Description of work

#### Summary of work

Fixes #36065. 

#### Further detail of work

This PR implements support for loading legacy file format in DNS Reduction GUI. More specifically:
- The neutron polarisation channel is determined automatically using the values of "currents" provided in the `B-fields` section of a legacy file
- Legacy filenames can be handled properly at the script generation stage

### To test:

- Download test data:
  - Present format: [Elastic_Powder_New.zip](https://github.com/mantidproject/mantid/files/13298995/Elastic_Powder_New.zip)
  - Legacy format: [Elastic_Powder_Legacy.zip](https://github.com/mantidproject/mantid/files/13299004/Elastic_Powder_Legacy.zip)
- Open the reduction GUI [Interfaces->Direct->DNS Reduction]

  Repeat the steps below for present and legacy format files and make sure that the outcome is identical  
- Browse to the directory where the test data are located using the `"Paths"` tab of the GUI
- Load and select the data for reduction using the `"Data"` tab
- Choose the data reduction options using the `"Options"` tab
- Reduce the data and generate a data reduction script using the `"Script Generator"` tab
- Visualize the results using the `"Plotting"` tab 





---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.
